From: Ondrej Jirman <megous@megous.com>
Date: Wed, 22 Sep 2021 19:30:39 +0200
Subject: arm64: dts: rk3399-pinephone-pro: Fixup DT validation issues

Fixes:

rk3399-pinephone-pro.dts:751.4-14: Warning (reg_format): /mipi@ff960000/panel/port:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
rk3399-pinephone-pro.dtb: Warning (pci_device_reg): Failed prerequisite 'reg_format'
rk3399-pinephone-pro.dtb: Warning (pci_device_bus_num): Failed prerequisite 'reg_format'
rk3399-pinephone-pro.dtb: Warning (i2c_bus_reg): Failed prerequisite 'reg_format'
rk3399-pinephone-pro.dtb: Warning (spi_bus_reg): Failed prerequisite 'reg_format'
rk3399-pinephone-pro.dts:747.8-756.5: Warning (avoid_default_addr_size): /mipi@ff960000/panel/port: Relying on default #address-cells value
rk3399-pinephone-pro.dts:747.8-756.5: Warning (avoid_default_addr_size): /mipi@ff960000/panel/port: Relying on default #size-cells value
rk3399-pinephone-pro.dts:747.8-756.5: Warning (graph_port): /mipi@ff960000/panel/port: graph node unit address error, expected "0"
rk3399-pinephone-pro.dts:748.4-25: Warning (graph_port): /mipi@ff960000/panel/port:#address-cells: graph node '#address-cells' is -1, must be 1
rk3399-pinephone-pro.dts:749.4-22: Warning (graph_port): /mipi@ff960000/panel/port:#size-cells: graph node '#size-cells' is -1, must be 0

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 0c213f8..51a7afa 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -612,7 +612,6 @@ fusb0: typec-portc@22 {
 		pinctrl-names = "default";
 		pinctrl-0 = <&fusb0_int>;
 		vbus-supply = <&vcc5v0_typec>;
-		status = "okay";
 
 		connector {
 			compatible = "usb-c-connector";
@@ -677,23 +676,25 @@ accelerometer@68 {
 			"0", "0", "1";
 	};
 
-	lis3mdl: magnetometer@1e {
+	lis3mdl: magnetometer@1c {
 		compatible = "st,lis3mdl-magn";
 		reg = <0x1c>;
 		interrupt-parent = <&gpio1>;
 		interrupts = <RK_PA1 IRQ_TYPE_LEVEL_LOW>;
 		vdd-supply = <&vcc3v3_sys>;
 		vddio-supply = <&vcc_1v8>;
+		status = "disabled";
 	};
 
 	// not populated currently
-	ak09911: compass@0c {
+	ak09911: compass@c {
 		compatible = "asahi-kasei,ak09911";
 		reg = <0x0c>;
 		interrupt-parent = <&gpio1>;
 		interrupts = <RK_PA1 IRQ_TYPE_LEVEL_LOW>;
 		vdd-supply = <&vcc_3v0>;
 		vid-supply = <&vcc_1v8>;
+		status = "disabled";
 	};
 };
 
@@ -711,6 +712,8 @@ &mipi_dsi {
 
 	ports {
 		mipi_out: port@1 {
+			#address-cells = <0>;
+			#size-cells = <0>;
 			reg = <1>;
 
 			mipi_out_panel: endpoint {
@@ -719,7 +722,7 @@ mipi_out_panel: endpoint {
 		};
 	};
 
-	panel {
+	panel@0 {
 		compatible = "hannstar,hsd060bhw4";
 		reg = <0>;
 		backlight = <&backlight>;
@@ -730,11 +733,6 @@ panel {
 		pinctrl-0 = <&display_rst_l>;
 
 		port {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			reg = <0>;
-
 			mipi_in_panel: endpoint {
 				remote-endpoint = <&mipi_out_panel>;
 			};
